subsys/ipc/ipc_service/ipc_service.c

```diff
@@ -14,6 +14,27 @@
 
 LOG_MODULE_REGISTER(ipc_service, CONFIG_IPC_SERVICE_LOG_LEVEL);
 
+int ipc_service_open_instance(const struct device *instance)
+{
+	const struct ipc_service_backend *backend;
+
+	if (!instance) {
+		LOG_ERR("Invalid instance");
+		return -EINVAL;
+	}
+
+	backend = (const struct ipc_service_backend *) instance->api;
+
+	if (!backend) {
+		LOG_ERR("Invalid backend configuration");
+		return -EIO;
+	}
+
+	__ASSERT(backend->open_instance != NULL, "open_instance hook not implemented");
```

Comment:
What if asserts are disabled for given build? I think we should handle `if (!backend->open_instance) return SOME_ERROR;` alongside `__ASSERT`.

Reply:
Yes, I'm a bit conflicted about this considering what @arnopo said in https://github.com/zephyrproject-rtos/zephyr/pull/40365#discussion_r751501726

Reply:
Perhaps following code would be a compromize
```
if( !backend->open_instance)
     return 0;

return backend->open_instance(instance); 
```
The most important is to not return an error if not defined

Reply:
ACK